src/read.rs

```diff
@@ -270,7 +270,7 @@ impl<R: Read> SaveReader<R> {
         let material_count = cmp::max(header2.materials.len(), 2);
         let physical_material_count = cmp::max(header2.physical_materials.len(), 2);
 
-        let mut bricks = vec![];
+        let mut bricks = Vec::with_capacity(header1.brick_count as usize);
```

Comment:
To address your pitfall: it might be worth adding a `const MAX_BRICK_COUNT_CAPACITY: usize` to this module that dictates the maximum reasonable amount of bricks to allocate when reading. Maybe 2,000,000 or so. Then, you could `with_capacity` the maximum between the save-specified brick count and that constant.

Reply:
Also, if you're concerned about too much being allocated for values *less* than this, consider [`shrink_to_fit`](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.shrink_to_fit) after the vector is populated.